### PR TITLE
fix: ignore byte array fields and add ddwaf:ignore struct tag

### PR DESCRIPTION
**Pull request by eliottness**

### CODE REVIEW

File: context.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ func (d RunAddressData) isEmpty() bool {
 // matches as a JSON string (usually opaquely used) along with the corresponding actions in any. In case of an error,
 // matches and actions can still be returned, for instance in the case of a timeout error. Errors can be tested against
 // the RunError type.
+// Struct fields having the tag `ddwaf:ignore` will not be encoded and sent to the WAF
```
Comment (RomainMuller) [inline context.go:73]: That should be `ddwaf:"ignore"`, I think. That feature also appears untested... SMH.

Comment (eliottness) [reply context.go:73]: Oops, forgot to include them, here they are

Comment (RomainMuller) [reply context.go:73]: Still,

```suggestion
// Struct fields having the tag `ddwaf:"ignore"` will not be encoded and sent to the WAF
```

**Review by RomainMuller — APPROVED** [review]

**Review by RomainMuller — APPROVED** [review]

**Review by RomainMuller — APPROVED** [review]

**Pull request merged**